Snap_rounding_2/include/CGAL/Float_snap_rounding_2.h

```diff
@@ -0,0 +1,452 @@
+#define CGAL_FLOAT_SNAP_ROUNDING_2_H
+
+
+#include <iostream>
+#include <CGAL/intersection_2.h>
+#include <CGAL/Surface_sweep_2_algorithms.h>
+#include <CGAL/Box_intersection_d/Box_d.h>
+#include <CGAL/box_intersection_d.h>
+#include <set>
+#include <vector>
+
+namespace CGAL {
+namespace Box_intersection_d {
+
+template<class NT_, int N>
+class Box_with_index_d: public Box_d< NT_, N, ID_EXPLICIT> {
+protected:
+    size_t m_index;
+public:
+    typedef Box_d< NT_, N, ID_EXPLICIT> Base;
+    typedef NT_                      NT;
+    typedef size_t                   ID;
+
+    Box_with_index_d() {}
+    Box_with_index_d( ID i) : m_index(i) {}
+    Box_with_index_d( bool complete, ID i): Base(complete), m_index(i) {}
+    Box_with_index_d(NT l[N], NT h[N], ID i) : Base( l, h), m_index(i) {}
+    Box_with_index_d( const Bbox_2& b, ID i) : Base( b), m_index(i) {}
+    Box_with_index_d( const Bbox_3& b, ID i) : Base( b), m_index(i) {}
+    ID  index() const { return m_index; }
+    // ID  id() const { return m_index; }
+};
+
+// Generic template signature of boxes, specialized for ID_FROM_HANDLE policy
+#if 0
+template<class NT_, int N, class Handle_, class IdPolicy = ID_FROM_INDEX>
+class Box_with_index_d : public Box_d< NT_, N, IdPolicy> {
+protected:
+    size_t m_index;
+public:
+    typedef Box_d< NT_, N, IdPolicy> Base;
+    typedef NT_                      NT;
+    typedef size_t                   ID;
+
+    Box_with_index_d() {}
+    Box_with_index_d( ID i) : m_index(i) {}
+    Box_with_index_d( bool complete, ID i): Base(complete), m_index(i) {}
+    Box_with_index_d(NT l[N], NT h[N], ID i) : Base( l, h), m_index(i) {}
+    Box_with_index_d( const Bbox_2& b, ID i) : Base( b), m_index(i) {}
+    Box_with_index_d( const Bbox_3& b, ID i) : Base( b), m_index(i) {}
+    ID  index() const { return m_index; }
+};
+
+// Specialization for ID_FROM_INDEX policy
+template<class NT_, int N>
+class Box_with_index_d<NT_, N, Handle_, ID_FROM_INDEX>
+    : public Box_d< NT_, N, ID_NONE> {
+protected:
+    size_t m_index;
+public:
+    typedef Box_d< NT_, N, ID_NONE> Base;
+    typedef NT_                      NT;
+    typedef size_t                   ID;
+
+    Box_with_index_d() {}
+    Box_with_index_d( ID i) : m_index(i) {}
+    Box_with_index_d( bool complete, ID i): Base(complete), m_index(i) {}
+    Box_with_index_d(NT l[N], NT h[N], ID i) : Base( l, h), m_index(i) {}
+    Box_with_index_d( const Bbox_2& b, ID i) : Base( b), m_index(i) {}
+    Box_with_index_d( const Bbox_3& b, ID i) : Base( b), m_index(i) {}
+    ID  index() const { return m_index; }
+    ID  id() const { return m_index; }
+};
+#endif
+}
+
+template <class PointsRange , class PolylineRange>
+void double_snap_rounding_2_disjoint(PointsRange &pts,
+		                                 PolylineRange &polylines)
+{
+  using Point_2 = std::remove_cv_t<typename std::iterator_traits<typename PointsRange::iterator>::value_type>;
+  using Kernel = typename Kernel_traits<Point_2>::Kernel;
```

Comment:
does it make sense to have named parameters for the traits? + point map maybe? Also we will need a traits concept with all the requirements of the function

Reply:
If the traits is a parameter you won't a point map (can be integrated in a model of the concept of the traits), so no NP for now I'd say.